### PR TITLE
Neel/updating page styles for mobile

### PR DESCRIPTION
**Pull request by neelismail01**
### Notion ticket link

[Start Mobile Responsive Designs for static pages](https://www.notion.so/uwblueprintexecs/Start-Mobile-Responsive-Designs-for-static-pages-71b40c3d94f1448ab3fde0ca1d376fb8)

### Implementation description

- Implemented the useBreakPointValue hook to determine device type and added logic to various components to appropriately style based on the recognized device type

### Steps to test

1. Right click on the webpage in the Chrome web browser, and click "Inspect"
2. Change the device type to mobile in the device toolbar
3. Verify that the webpages match the Figma designs when in the mobile view

#### Videos for reference during testing:
**Document Upload Page:**

https://user-images.githubusercontent.com/23081661/141712591-4185d423-448e-4fdf-8726-ec7503493e62.mov

**Home Page:**

https://user-images.githubusercontent.com/23081661/141713281-5ad89202-d19d-4dc2-b43d-c4f00559a4c5.mov

**Login Page:**

https://user-images.githubusercontent.com/23081661/141712667-fff30149-1748-4b97-979b-62906db651b7.mov

### What should reviewers focus on?

- Ensure that the webpages match the Figma designs

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/login.tsx [hunk 1]
```diff
@@ -57,7 +62,7 @@ export default function Login(): JSX.Element {
                     <FormControl id="email">
```
Comment (MatoPlus) [inline pages/login.tsx:67]: I think some of the text should be center-aligned: 


For line 52
```typescript
                        <Text
                            fontWeight="400"
                            fontSize="16px"
                            mt="20px"
                            textAlign="center"
                        >
                            Registration for Summer 2021 classes begin June 31,
                            2021
                        </Text>
```

image

File: pages/login.tsx [hunk 2]
```diff
@@ -126,14 +131,14 @@ export default function Login(): JSX.Element {
                         <Text
                             fontWeight="400"
                             fontSize="14px"
-                            mt="70px"
+                            mt="50px"
                             color="brand.300"
                         >
```
Comment (MatoPlus) [inline pages/login.tsx:142]:
```suggestion
                            color="brand.300"
                            textAlign="center"
                        >
```
same for this text

File: src/components/WelcomeToSDC.tsx [hunk 3]
```diff
@@ -73,21 +77,25 @@ export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
                                 )}
```
Comment (MatoPlus) [inline src/components/WelcomeToSDC.tsx:82]: For the button, please add a breakpoint for width on line 60, (change on line 70)

```typescript
                                        <Button
                                            color="white"
                                            backgroundColor={
                                                colourTheme.colors.Blue
                                            }
                                            _hover={{
                                                backgroundColor:
                                                    colourTheme.colors
                                                        .LightBlue,
                                            }}
                                            width={{ base: "100%", md: "50%" }}
                                            borderRadius="6px"
                                            fontWeight={"200"}
                                        >
                                            {t("home.registerNow")}
                                        </Button>
```

image

File: src/components/WelcomeToSDC.tsx [hunk 4]
```diff
@@ -20,6 +21,7 @@ type WelcomeToSDCProps = {
 
 export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
     const { t } = useTranslation("common");
+    const variant = useBreakpointValue({ base: "mobile", lg: "computer" });
```
Comment (MatoPlus) [inline src/components/WelcomeToSDC.tsx]: you got the idea here but can we maybe use a boolean, this seems difficult to keep track of
```suggestion
    const isMobileLayout = useBreakpointValue({ base: true, lg: false });
```

File: src/components/WelcomeToSDC.tsx [hunk 5]
```diff
@@ -44,10 +46,12 @@ export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
             <TabPanels>
                 <TabPanel>
                     <HStack spacing="24px">
-                        <Box w="50%">
+                        <Box w={variant === "computer" ? "50%" : "100%"}>
```
Comment (MatoPlus) [inline src/components/WelcomeToSDC.tsx]: do we need a break point here?

File: src/components/WelcomeToSDC.tsx [hunk 6]
```diff
@@ -73,21 +77,25 @@ export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
...
+                                    </Box>
+                                </Flex>
+                            </Box>
+                        )}
                     </HStack>
                 </TabPanel>
                 <TabPanel>
                     <HStack spacing="24px">
-                        <Box w="50%">
+                        <Box w={variant === "computer" ? "50%" : "100%"}>
```
Comment (MatoPlus) [inline src/components/WelcomeToSDC.tsx]: do we need to use the breakpoint state here?

File: src/components/Navbar.tsx [hunk 7]
```diff
@@ -37,25 +55,141 @@ const NavLink = ({ href, children }: { href: string; children: ReactNode }) => (
...
+            </DrawerContent>
+        </Drawer>
+    );
+};
+
 export const DEFAULT_NAVBAR_HEIGHT = 16;
 
 export const Navbar: React.FC<NavbarProps> = (props) => {
     const router = useRouter();
+    const variant = useBreakpointValue({ base: "sidebar", lg: "navbar" });
```
Comment (MatoPlus) [inline src/components/Navbar.tsx]: let's stick with booleans:
```suggestion
    const isSideBar = useBreakpointValue({ base: true, lg: false });
```

**Review by MatoPlus — CHANGES_REQUESTED** [review]
Looks really good! I caught some small things we can change.

**Review by bekahma — APPROVED** [review]

**Pull request merged**